crates/uv-resolver/src/pubgrub/report.rs

```diff
@@ -725,12 +725,18 @@ impl PubGrubReportFormatter<'_> {
                     .skip_while(|url| *url != found_index)
                     .nth(1)
                 {
-                    hints.insert(PubGrubHint::UncheckedIndex {
-                        package: package.clone(),
-                        range: set.clone(),
-                        found_index: found_index.clone(),
-                        next_index: next_index.clone(),
-                    });
+                    // Do not include the hint if the set is "all versions"
+                    if !set
+                        .iter()
+                        .all(|range| matches!(range, (Bound::Unbounded, Bound::Unbounded)))
```

Comment:
What is this testing for exactly? Sorry!

Reply:
An unbounded range, e.g., `foo` instead of `foo>=0.1`

Reply:
So if it's unbounded, does that mean we didn't find any versions at all?

Reply:
Yeah, see the error message in the linked issue. It doesn't make sense to say that a subsequent index may include the version you've requested.